mbed_targets/mbed_targets.py

```diff
@@ -57,15 +57,11 @@ def __init__(self) -> None:
         Raises:
             TargetDatabaseError: Failed to get the target data.
         """
-        self._target_data: List[dict] = target_database.get_target_data()
+        target_data = target_database.get_target_data()
+        self._all_targets = {target.product_code: target for target in [MbedTarget(target) for target in target_data]}
```

Comment:
Wow, that's quite a statement right there. Can this be simplified?

Reply:
Might be clearer as a plain `for` loop rather than a nested comprehension.